docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,250 @@
+[[fleet-agent-proxy-support]]
+= Use a proxy server with {agent} and {fleet}
+
+Your organization’s security strategy and other considerations may require you
+to use a proxy server between some components in your deployment. Support is
+available in {agent} and {fleet} for connections through HTTP Connect (HTTP 1
+only) and Socks5 proxy servers.
+
+//QUESTION: Do we want to provide a list of popular proxy servers that will work?
+
+This guide describes where a proxy server is allowed in your deployment and how
+to configure proxy settings for {agent} and {fleet}. The steps for deploying the
+proxy server itself are beyond the scope of this article.
+
+NOTE: Some environments require users to authenticate with the proxy. There are
+no explicit settings for proxy authentication in {agent} or {fleet}, except the
+ability to pass credentials in the URL or as keys/tokens in headers, as
+described later.
+
+[discrete]
+[[elastic-agent-proxy-config]]
+== Configure {agent} proxy settings
+
+Configure proxy settings for {agent} when it must connect through a proxy server
+to:
+
+* Download artifacts from `artifacts.elastic.co` for subprocesses or binary
+upgrades
+* Send data to {es}
+* Retrieve agent policies from {fleet-server}
+* Retrieve agent policies from {es} (when running as a {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings in:
+
+* <<host-proxy-env-vars>>
+* <<es-output-proxy-settings>>
+* <<cli-proxy-settings>>
+
+[discrete]
+[[host-proxy-env-vars]]
+=== Host environment variables
+
+Set environment variables on the host to configure default proxy settings.
+The {agent} uses host environment settings by default if no proxy settings are
+specified elsewhere. You can override host proxy settings later when you
+configure the {agent} and {fleet} settings. The following environment variables
+are available on the host:
+
+|===
+|Variable |Description
+
+|`HTTP_PROXY`
+|URL of the proxy server for HTTP traffic. 
+
+|`HTTPS_PROXY`
+|URL of the proxy server for HTTPS traffic. `HTTPS_PROXY` takes precedence over
+`HTTP_PROXY` for https requests.
+
+|`NO_PROXY`
+|IP addresses or domain names that should not use the proxy. Supports patterns.
+|===
+
+The proxy URL can be a complete URL or `host[:port]`, in which case the `http`
+scheme is assumed. An error is returned if the value is a different form.
+
+Keep in mind that proxy environment settings may affect other components on the
+host that send HTTP requests.
+
+[discrete]
+[[es-output-proxy-settings]]
+=== Proxy settings in the {agent} policy
+
+Configure proxy settings in the {agent} policy to override proxy settings
+specified by environment variables. For example, you can configure proxy
+settings in the {es} output to override host environment settings.
+
+//QUESTION: Is there a way to set the proxy URL for an input?
```

Comment:
I have not tested this, but I would expect it to be done as part of the input configuration on the policy?

Reply:
@michel-laterman I couldn't find out where to do this. Who can help me with this question? Should I just not mention setting this in the input (if it's not tested, IMO we shouldn't document it).

Reply:
I think that someone who worked on an integration may know more?
I haven't done testing of this feature.

Reply:
@ruflin Do you know how to configure the proxy URL for an input? Of if that's possible? I'm not sure who else to ask.

Reply:
This depends on each input. Each input has its own configuration and likely in most cases a proxy config does not make sense. This is the same config that is used for modules, for example like here for the http module https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-http.html If this module / input does not support a proxy, it can't be used.

I don't think we should document anything around this here as it is very input specific and I don't see how it fits into the overall proxy topic.

Reply:
Thanks, Nicolas. I won't mention inputs in this topic.